controllers/vrg_volrep.go

```diff
@@ -719,13 +718,23 @@ func (v *VRGInstance) reconcileMissingVR(pvc *corev1.PersistentVolumeClaim, log
 
 	err := v.reconciler.Get(v.ctx, vrNamespacedName, volRep)
 	if err == nil {
+		if !volRep.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
A deleted VR in a prior loop changes its generation, and further processAsPrimary|Secondary will error out as the status generation and actual generation will mismatch.

As a result once deleted, wait for it to be deleted, than process it.

This was tricky to find as this function is very silent (logs), hence added some additional logs to the same.